src/libraries/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -1192,21 +1157,21 @@ private static unsafe int WriteFileNative(IntPtr hFile, ReadOnlySpan<byte> bytes
                         // [https://docs.microsoft.com/en-us/windows/console/writeconsole]
                         // However, we do not need to worry about that because the StreamWriter in Console has
                         // a much shorter buffer size anyway.
-                        int charsWritten;
-                        writeSuccess = Interop.Kernel32.WriteConsole(hFile, p, bytes.Length / BytesPerWChar, out charsWritten, IntPtr.Zero);
-                        Debug.Assert(!writeSuccess || bytes.Length / BytesPerWChar == charsWritten);
+                        if (Interop.Kernel32.WriteConsole(_handle, p, buffer.Length / BytesPerWChar, out int charsWritten, 0))
+                        {
+                            Debug.Assert(buffer.Length / BytesPerWChar == charsWritten);
+                            return;
+                        }
                     }
                 }
-                if (writeSuccess)
-                    return Interop.Errors.ERROR_SUCCESS;
 
                 // For pipes that are closing or broken, just stop.
                 // (E.g. ERROR_NO_DATA ("pipe is being closed") is returned when we write to a console that is closing;
                 // ERROR_BROKEN_PIPE ("pipe was closed") is returned when stdin was closed, which is not an error, but EOF.)
                 int errorCode = Marshal.GetLastPInvokeError();
                 if (errorCode == Interop.Errors.ERROR_NO_DATA || errorCode == Interop.Errors.ERROR_BROKEN_PIPE || errorCode == Interop.Errors.ERROR_PIPE_NOT_CONNECTED)
-                    return Interop.Errors.ERROR_SUCCESS;
-                return errorCode;
+                    return;
+                throw Win32Marshal.GetExceptionForWin32Error(errorCode);
```

Comment:
Most of the changes in this file are refactoring without a major clean win to me. Please revert them and keep only the changes that apply to the removal of unused `_isPipe`. Thank you!

Reply:
IMHO, removing the unnecessary ReadNative and WriteNative layer is good. It makes the code a bit faster, smaller and easier to follow.

Reply:
> It makes the code a bit faster, smaller and easier to follow.

OK, let's keep the proposed changes.